mail_activity_reminder/models/mail_activity.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2020 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from pytz import timezone, UTC
+
+from odoo import _, api, fields, models
+
+
+class MailActivity(models.Model):
+    _inherit = 'mail.activity'
+
+    last_reminder = fields.Date()
+
+    @api.model
+    def _get_activities_to_remind(self):
+        utc_now = self.env.context.get(
+            'mail_activity_reminder_now',
+            fields.Datetime.now()
+        ).replace(tzinfo=UTC)
+
+        activities_to_remind = self.env['mail.activity']
+        for activity in self.search([]):
+            tz = activity.user_id.sudo().tz or 'UTC'
+
+            local_today = utc_now.astimezone(timezone(tz)).date()
+            local_deadline = activity.date_deadline
+            if local_deadline < local_today:
+                continue
+
+            reminders = activity.activity_type_id._get_reminder_offsets()
+            if not reminders:
+                continue
+
+            reminder = (local_deadline - local_today).days
+            if reminder not in reminders:
+                continue
+
+            if activity.last_reminder \
+                    and activity.last_reminder >= local_today:
+                continue
+
+            activities_to_remind |= activity
```

Comment:
please elaborate a bit more, I didn't get it

Reply:
```py
to_remind_ids = []
....
to_remind_ids.append(activity.id)
...
return self.browse(to_remind_ids)
```

Reply:
won't that be less readable and more work? instead of working with recordsets?

Reply:
less readable? Why? In any case it boosts performance. Keep it in mind every time you think about using recordsets ;)

Reply:
but I already have all the recordset, why making a `browse` would be faster? It would be great to get a link to code with the "why it's bad to use recordsets" part :D

Reply:
no, you don't have the recordset. You create an empty one at line 17. The "why" is: it's slower :wink: 
We knew it since a while but Guewen recently did a test, see here https://twitter.com/guewenb/status/1196775451739459584

Reply:
Got it, curious unexpected behaviour. I've updated code a bit differently